sources/api/apiclient/src/uri_resolver.rs

```diff
@@ -0,0 +1,846 @@
+            .await
+            .context(StdinReadSnafu)?;
+        Ok(buf)
+    }
+}
+
+/// Uri Resolver that reads from a local file.
+///
+/// This resolver accepts URIs of the form `file:///path/to/file` (or on Windows
+/// `file://C:/path/to/file`), converts them to a `PathBuf`, and returns the
+/// file’s entire contents as a UTF-8 string.
+pub struct FileUri {
+    path: PathBuf,
+}
+
+impl TryFrom<&SettingsInput> for FileUri {
+    type Error = ResolverError;
+    fn try_from(input: &SettingsInput) -> ResolverResult<Self> {
+        use resolver_error::*;
+
+        let url = input.parsed_url.clone().context(FileUriSnafu {
+            input_source: input.input.clone(),
+        })?;
+
+        // only accept file://
+        ensure!(
+            url.scheme() == "file",
+            FileUriSnafu {
+                input_source: url.to_string()
+            }
+        );
+
+        // convert to PathBuf or error
+        let path = url.to_file_path().ok().context(FileUriSnafu {
+            input_source: url.to_string(),
+        })?;
+
+        Ok(FileUri { path })
+    }
+}
+
+#[async_trait]
+impl UriResolver for FileUri {
+    async fn resolve(&self) -> ResolverResult<String> {
+        use resolver_error::*;
+        tokio::fs::read_to_string(&self.path)
+            .await
+            .context(FileReadSnafu {
+                input_source: self.path.to_string_lossy().into_owned(),
+            })
+    }
+}
+
+/// Uri Resolver that fetches over HTTP(S).
+///
+/// This resolver accepts URIs beginning with `http://` or `https://` and
+/// performs a GET request with `reqwest`, returning the response body as a
+/// UTF-8 string (erroring on non-2xx status).
+pub struct HttpUri {
+    url: Url,
+}
+
+impl TryFrom<&SettingsInput> for HttpUri {
+    type Error = ResolverError;
+    fn try_from(input: &SettingsInput) -> ResolverResult<Self> {
+        use resolver_error::*;
+        let url = input.parsed_url.clone().context(InvalidHTTPUriSnafu {
+            input_source: input.input.clone(),
+        })?;
+        ensure!(
+            url.scheme() == "http" || url.scheme() == "https",
+            InvalidHTTPUriSnafu {
+                input_source: url.to_string()
+            }
+        );
+        Ok(HttpUri { url })
+    }
+}
+
+#[async_trait]
+impl UriResolver for HttpUri {
+    async fn resolve(&self) -> ResolverResult<String> {
+        use resolver_error::*;
+        const MAX_SIZE_BYTES: u64 = 100 * 1024 * 1024;
+        // 1) issue the GET
+        let resp = reqwest::get(self.url.clone())
+            .await
+            .context(HttpRequestSnafu {
+                uri: self.url.to_string(),
+            })?;
+
+        // 2) check status
+        let resp = if resp.status().is_success() {
+            resp
+        } else {
+            return HttpStatusSnafu {
+                uri: self.url.to_string(),
+                status: resp.status(),
+            }
+            .fail();
+        };
```

Comment:
`resp` isn't actually changed in this, so it's idiomatically/stylistically better written with `ensure!`

```suggestion
        ensure!(resp.status().is_success(), HttpStatusSnafu {
                uri: self.url.to_string(),
                status: resp.status(),
            }
        );
```

Reply:
Resolved - replaced `if, else` with `ensure` since `resp` stays the same.